provisioner/clusterpy.go

```diff
@@ -219,16 +220,18 @@ func (p *clusterpyProvisioner) provision(
 	}
 
 	// find the best subnet for each AZ
-	azInfo := selectSubnetIDs(subnets)
+	azInfoLBs := selectSubnetIDs(subnets, subnetELBRoleTagName)
+	azInfoWorkers := selectSubnetIDs(subnets, subnetWorkerRoleTagName)
 
 	// if availability zones are defined, filter the subnet list
 	if azNames, ok := cluster.ConfigItems[availabilityZonesConfigItemKey]; ok {
-		azInfo = azInfo.RestrictAZs(strings.Split(azNames, ","))
+		azInfoLBs = azInfoLBs.RestrictAZs(strings.Split(azNames, ","))
+		azInfoWorkers = azInfoWorkers.RestrictAZs(strings.Split(azNames, ","))
 	}
 
 	// TODO legacy, remove once we switch to Values in all clusters
```

Comment:
> // TODO legacy, remove once we switch to Values in all clusters

Does this comment somehow change meaning or is it maybe outdated?

Reply:
True, this is being used for controlled migrations between subnets. Let me update the comment as well.